### DISCUSSION TITLE
Measuring data on a truffle project

### DISCUSSION THREAD

**Discussion by ryurt46** [ConsenSys-archive/truffle#6030] [General Q&A]
Hi I'm working on a smart contract in ganache and would like to do some data measurement, e.g how much data each deployment needs and etc.

Thanks :)

**Comment by FlaviuWeb3** [comment]
Hi,

To measure the data consumption of your smart contract deployment in Ganache, you can follow these steps:

Set up Ganache: Make sure you have Ganache installed and running on your local machine. Ganache provides a local blockchain environment for testing and development.

Write and compile your smart contract: Create your smart contract using Solidity or any other supported language. Once your contract is ready, compile it using a Solidity compiler such as solc or the compiler provided by your development framework.

Deploy your smart contract: Use a development framework like Truffle to deploy your smart contract to the Ganache network.

Monitor transaction data: Ganache provides a user interface that displays information about each transaction, including the amount of gas consumed. Gas consumption is a good indicator of the data storage requirements because every data storage operation, such as writing to a contract's storage variables, consumes gas.

After deploying your smart contract, open the Ganache UI.
You should see a list of transactions. Locate the transaction corresponding to your smart contract deployment.
Click on that transaction to view its details.
Look for the "Gas Used" field, which represents the amount of gas consumed during the deployment. This gives you an estimate of the data storage requirements.
